Plotting Int64 columns with nulled integers (NAType) fails #32073

jeandersonbc opened the pull request:
- [x] closes #32073
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

jeandersonbc committed:
BUG: fixes unhandled NAType when plotting (#32073)

jeandersonbc committed:
BUG: fixes unhandled NAType when plotting (#32073)

jeandersonbc committed:
Fixed bad formatting

MarcoGorelli commented:
Thanks @jeandersonbc 

However, before this PR can be accepted, you need to write tests / make sure you don't break existing ones - see [contributing to the code base](https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#contributing-to-the-code-base):

> pandas is serious about testing and strongly encourages contributors to embrace test-driven development (TDD). This development process “relies on the repetition of a very short development cycle: first the developer writes an (initially failing) automated test case that defines a desired improvement or new function, then produces the minimum amount of code to pass that test.” So, before actually writing any code, you should write your tests. Often the test can be taken from the original GitHub issue. However, it is always worth considering additional use cases and writing corresponding tests.
> 
> Adding tests is one of the most common requests after code is pushed to pandas. Therefore, it is worth getting in the habit of writing tests ahead of time so this is never an issue.

jbrockmendel committed:
REF: collect+parametrize reorder_levels tests (#32373)

jeandersonbc commented:
Thanks for the feedback @MarcoGorelli! I'm wondering whether the example provided in the related issue (first comment in https://github.com/pandas-dev/pandas/issues/32073) would be sufficient to be added as a test in the `TestDataFramePlots` (test_frame.py) class. I started the discussion on tests on the issue yesterday (although here might be more appropriated).

rushabh-v committed:
TST: Allow definition of `pd.CategoricalDtype` with a specific `categories.dtype` (#32115)

jbrockmendel committed:
TYP: annotations for internals, set_axis (#32376)

jbrockmendel committed:
misplaced DataFrame.join test (#32375)

TomAugspurger commented on pandas/plotting/_matplotlib/core.py:
```diff
@@ -413,7 +413,13 @@ def _compute_plot_data(self):
+            values = numeric_data[col]
+            if values.isna().any().all():
+                values = values.astype(float)
+
```
It seems like date-like data can be included here, and we don't want to convert those to floats.

I think this should be restricted to

```python
if is_integer_dtype(values.dtype):
    values = values.to_numpy(dtype="float", na_value=np.nan)
```

TomAugspurger reviewed (commented):
Thanks for working on this. We'll need a test and a release note in 1.1.0.rst.

Iqrar99 committed:
DOC: Fixed ES01, PR07, SA04 error in pandas.core.groupby.DataFrameGroupBy.shift (#32356)

TomAugspurger committed:
BUG: Pickle NA objects (#32104)

According to
https://docs.python.org/3/library/pickle.html#object.__reduce__,

> If a string is returned, the string should be interpreted as the name
> of a global variable. It should be the object’s local name relative to
> its module; the pickle module searches the module namespace to determine
> the object’s module. This behaviour is typically useful for singletons.

Closes https://github.com/pandas-dev/pandas/issues/31847

AdrianMastronardi committed:
DOC: Fix SA04 errors in docstrings #28792 (#32182)

jbrockmendel committed:
CLN: remove _igetitem_cache (#32319)

jbrockmendel committed:
Avoid unnecessary values_from_object (#32398)

jbrockmendel committed:
ENH: infer freq in timedelta_range (#32377)

jbrockmendel committed:
BUG: 2D DTA/TDA arithmetic with object-dtype (#32185)

jbrockmendel committed:
TST: broken off from #32187 (#32258)


Co-authored-by: Simon Hawkins <simonjayhawkins@gmail.com>

jbrockmendel committed:
REF: simplify PeriodIndex._shallow_copy (#32280)

jbrockmendel committed:
CLN: setitem_with_indexer cleanups (#32341)

jbrockmendel committed:
BUG: None / Timedelta incorrectly returning NaT (#32340)

SaturnFromTitan committed:
TST: Using  more fixtures in of tests/base/test_ops.py (#32313)

jbrockmendel committed:
CLN: remove unused values from interpolate call (#32400)

ShaharNaveh committed:
CLN: some code cleanups to pandas/_libs/missing.pyx (#32367)

s-scherrer committed:
BUG: fixes bug when using sep=None and comment keyword for read_csv (#31667)

dsaxton committed:
Don't create _join_functions (#32336)

a-y-khan committed:
API: replace() should raise an exception if invalid argument is given (#31946)

dsaxton committed:
BUG: Fix __ne__ comparison for Categorical (#32304)

simonjayhawkins committed:
CLN: clean-up show_versions and consistently use null for json output (#32042)

Commit:
Add missing newline (#32404)

jeandersonbc committed:
Added simple test case

jeandersonbc commented:
well, I'm going to close this pull request and work in a single branch as this was not the first time that I messed up after updating my branch
